app/src/main/java/com/github/geohunt/app/ui/components/CreateNewChallenge.kt

```diff
@@ -0,0 +1,190 @@
+                painter = bitmapPainter,
+                modifier = Modifier
+                    .aspectRatio(bitmapPainter.intrinsicSize.width / bitmapPainter.intrinsicSize.height)
+                    .fillMaxSize(),
+                contentDescription = "Photo just taken of the challenge"
+            )
+
+            Spacer(Modifier.height(25.dp))
+
+            LinkText(listOf(
+                LinkTextData("By creating a challenge, you agree to GeoHunt's "),
+                LinkTextData(
+                    text = "Community Guidelines",
+                    tag = "",
+                    annotation = "",
+                    onClick = {
+                        context.startActivity(openCommunityGuidelinesIntent)
+                    }
+                )
+            ), style = Typography.h5)
+
+            Spacer(Modifier.height(15.dp))
+
+            Button(
+                onClick = {
+                          database.createChallenge(
+                              thumbnail = bitmap,
+                              location = locationRequest.lastLocation.value!!,
+                              expirationDate = null
+                          ).toCompletableFuture(activity)
+                              .thenApply(onChallengeCreated)
+                              .thenApply {  }
+                              .exceptionally(onFailure)
+                },
+                enabled = (locationRequest.lastLocation.value != null)
+            ) {
+                Text("Create challenge")
+            }
+        }
+    }
+}
+
+@Composable
+fun CreateNewChallenge(
+    database: Database,
+    onChallengeCreated: (Challenge) -> Unit = {},
+    onFailure: (Throwable) -> Unit = {}
+) {
+    val context = LocalContext.current
+    val file = remember { context.createImageFile() }
+    val uri = remember {
+        FileProvider.getUriForFile(
+            context,
+            BuildConfig.APPLICATION_ID + ".provider",
+            file
+        )
+    }
+    val bitmapResult = remember { mutableStateOf<Bitmap?>(null) }
+    val permissions = rememberPermissionsState(Manifest.permission.CAMERA)
+    val cameraLauncher = rememberLauncherForActivityResult(
+        ActivityResultContracts.TakePicture()
+    ) {
+        Log.i("GeoHunt", "Returning from camera")
+        if (!it) {
+            onFailure(RuntimeException("Failed to take the photo"))
+        }
+        else {
+            BitmapUtils.loadFromFileAsync(file).asTask()
+                .thenDo {  bitmap ->
+                    Log.i("GeoHunt", "Resizing the bitmap...")
+                    BitmapUtils.resizeBitmapToFitAsync(bitmap, R.integer.maximum_number_of_pixel_per_photo).asTask()
+                }
+                .toCompletableFuture(context.findActivity())
+                .thenAccept { bitmap ->
+                    Log.i("GeoHunt", "Updating bitmapResult value")
+                    bitmapResult.value = bitmap
+                }
+                .thenApply {  }
```

Comment:
Is the thenApply function supposed to be blank here?

Reply:
Yes because the next exceptionally can then return a unit. I think it is more readable that way